docker-qgis/process_projectfile.py

```diff
@@ -1,40 +1,61 @@
+import io
 import logging
 from pathlib import Path
-from typing import Dict
+from typing import NamedTuple, Optional
 from xml.etree import ElementTree
 
-from qfieldcloud.qgis.utils import BaseException, get_layers_data, layers_data_to_string
+from qfieldcloud.qgis.utils import (
+    FailedThumbnailGenerationException,
+    InvalidFileExtensionException,
+    InvalidXmlFileException,
+    ProjectFileNotFoundException,
+    get_layers_data,
+    layers_data_to_string,
+)
 from qgis.core import QgsMapRendererParallelJob, QgsMapSettings, QgsProject
 from qgis.PyQt.QtCore import QEventLoop, QSize
 from qgis.PyQt.QtGui import QColor
 
 logger = logging.getLogger("PROCPRJ")
 
 
-class ProjectFileNotFoundException(BaseException):
```

Comment:
Exceptions were moved to utils.py to be able to catch them there without incurring any circular dependency.